chalicelib/lib/helpers.py

```diff
@@ -25,17 +28,64 @@ def date_range(start_date, stop_date):
 
 
 def validate_date(date, format_str) -> bool:
+    """
+    Validate a string in date format is valid
+    """
 
     try:
-        if ":" in date:
-            start, stop = date.split(":")
-            datetime.strptime(start, format_str)
-            datetime.strptime(stop, format_str)
-            if start > stop:
-                return False
-        else:
-            datetime.strptime(date, format_str)
-    except TypeError:
+        datetime.strptime(date, format_str)
+    except (TypeError, ValueError) as error:
+        log.error(f"Unable to validate date {date}. Error was: {error}")
         return False
 
     return True
+
+
+def parse_date(date: str, format_str: str = "%Y-%m-%d") -> dict:
+    """
+    Parse the date from a string.
+    Expects these formats:
+    "today"
+    "2019-01-01"
+    "2019-01-01:2019-01-02"
+    """
+
+    dates = dict()
+    if date == "today":
+        dates["today"] = datetime.now()
+
+    if ":" in date:
+        first_date, second_date = date.split(":")
+
+        if validate_date(first_date, format_str=format_str) and validate_date(second_date, format_str=format_str):
+            if first_date > second_date:
+                log.error(f"First date {first_date} needs to be smaller than second date {second_date}")
+                return dates
+            dates[f"{first_date}"] = datetime.strptime(first_date, format_str)
+            dates[f"{second_date}"] = datetime.strptime(second_date, format_str)
+    else:
+        if validate_date(date, format_str=format_str):
+            dates[f"{date}"] = datetime.strptime(date, format_str)
+
+    return dates
+
+
+def get_dates(first_date: datetime, second_date: datetime = None, format_str: str = "%Y-%m") -> list:
```

Comment:
I guess this is not the best function name. Any suggestion on how to make this more clear?

Reply:
The docstring is quite missleading, since it supports a single date. Will update it.

Reply:
This is really about months

Reply:
@tommymalmqvist well since you could pass a different format string this function could extract for instance only years as well. Tried to make it a bit modular

Reply:
It’s hardcoded to use %Y-%m

Reply:
It has a default value, but could be changed if the caller specify it

Reply:
Line 89 hard codes to %Y-%m for the dates in the returned list. Can we change the function name so that can be inferred?

Reply:
Yes good catch

Reply:
@tommymalmqvist I pushed an update to resolve the hardcoded format. What about the name of the function?

Is one of these better?
* filter_dates
* get_matching_dates

Other suggestion?